### ISSUE TITLE
[NEED ASSIST] Protocol requirement cannot be declared '@_spi'

### ISSUE THREAD

**Issue by norbertlorinczi** [Adyen/adyen-ios#1130]
Hello!

I'm trying to import the Adyen framework v5.0 via Cocoapods, like it's described [in the installation guide](https://adyen.github.io/adyen-ios/5.0.0/documentation/adyen/installation).
The problem I'm facing is at the compilation phase, I'm receiving some '@_spi' related errors:
"Protocol requirement 'context' cannot be declared '@_spi' without a default implementation in a protocol extension"

<img width="1270" alt="Screenshot 2023-01-26 at 15 31 42" src="https://user-images.githubusercontent.com/60603222/214848066-e0d7bfb4-1fb9-452a-8ba3-bd0440b6d348.png">

And it is the same for the `checkoutAttemptId` from `var PaymentMethodDetails`, and for the `func buildComponent` func from `PaymentMethod`.

Have I missed something?
Any help or additional information is appreciated! Thank you!

**Comment by erenbesel** [comment]
Hey @norbertlorinczi 
`@_spi(AdyenInternal)` is meant for access level among modules in the SDK. It is not meant for external access. 

Are you trying to access an @_spi property/function? 

You can see the sample app [IntegrationExample](https://github.com/Adyen/adyen-ios/blob/develop/Demo/Common/IntegrationExample.swift) for more info.

**Comment by norbertlorinczi** [comment]
No, I'm not trying to access any @_spi, it just says that those properties/function are missing a default implementation and it cannot be declared @_spi.
This occurs in the build phase even if there is no Adyen related code present. I tried it with a clean project, and just by adding the framework to the workspace via cocoapods the same failure happens.
I looked at the integration example that you linked, thank you, although the problem on my side seems to be before any use of the framework, the pod does not compile.

Am I missing something with the setup?

**Comment by erenbesel** [comment]
Hey @norbertlorinczi 
Sorry for the late reply. I'm having some issues with cocoapods on my machine. Once that is resolved, I will try with a fresh project and see if I'm getting similar issues.

**Comment by erenbesel** [comment]
Hey @norbertlorinczi 
Finally was able to resolve my pod issues and created a fresh project with cocoapods version 1.11.3 and installed our SDK via `pod 'Adyen'` and had no issue with building the project.

**Comment by norbertlorinczi** [comment]
Hi @erenbesel 
Thank you for your help!
It seems that the `BUILD_LIBRARY_FOR_DISTRIBUTION` flag is causing the issue in the build settings of the sdk. We are trying to use the Adyen SDK in our own SDK as a dependency, and the `BUILD_LIBRARY_FOR_DISTRIBUTION` flag is set to `YES` after the `pod install`, and we have to see if we can solve this issue. 
FYI, this happens only with version 5.0, the 4.10 version has no problem with that flag setting.

**Comment by descorp** [comment]
Hey @norbertlorinczi 

How does you `podspec` looks like?

**Comment by norbertlorinczi** [comment]
Hey @descorp 
So just to clarify, for a clean project works like this (It was my mistake that I did some post install changes why I tried it first):
```
platform :ios, '14.0'
source 'https://github.com/CocoaPods/Specs.git'
use_frameworks!

workspace 'Test'

target 'Test' do
  pod 'Adyen'
end
```
So this takes the last version (5.0) currently and if you just add it to the project it works, it has the `BUILD_LIBRARY_FOR_DISTRIBUTION ` flag in the build settings set to `NO`.
But in my case, the initial issue was that I have the Adyen SDK as a dependency in my sdk, so it is specified in the `podspec` file of my sdk just like this:
`s.dependency 'Adyen'`
And even without any post install changes the `BUILD_LIBRARY_FOR_DISTRIBUTION` is set to `YES` by cocoapods, since it's a dependency of another sdk, and in the build phase it does not find a default implementation for the aforementioned properties/functions marked with the `@_spi` tag. This is only the case if that flag is set to `YES`. 

Do you have any suggestions? 😄

**Comment by descorp** [comment]
Hey @norbertlorinczi 

Sorry, no solid solution. 
As you can probably guess, we haven't really explored the use cases of our SDK embedded in other SDKs 😅

I may suggest trying to use XCFramework(s) directly as `vendored_frameworks` 🤔
You can build one with `xcodebuild archive` or via Carthage.

**Comment by norbertlorinczi** [comment]
Closing issue since it was related to the project settings.

**Comment by paillou** [comment]
Hey @norbertlorinczi 

Were you able to find a workaround with this in the end?

I'm asking because I'm facing the same issue as I'm working on an app using development pods (acting as internal modules) and one of them contains the `Adyen` dependency in its podspec.

The solution proposed by @descorp will probably work but I'd like to try it as a last resort since it's not practical to handle SDK updates this way.

**Comment by norbertlorinczi** [comment]
Hi @paillou 

Sorry for the late reply. Well, for the time being we just disabled that flag in our project and not doing any postinstall changes  and it works that way. We have to see what will be the final solution and what implications that will have. 😄 
It is strange that when that `BUILD_LIBRARY_FOR_DISTRIBUTION` flag is on yes it encounters a build time issue, apparently because it just can't find some implementation tagged with @spi.

**Comment by paillou** [comment]
Thanks @norbertlorinczi 

On my side I found out why this flag is set to YES, it's due to another dependency which includes the following lines in its podpsec:
```ruby
 spec.xcconfig = {
    :BUILD_LIBRARY_FOR_DISTRIBUTION => 'YES'
  }
```
... but I don't know for what reason 🤔

I manage to overcome this using this `post_install` script in my Podfile:
```ruby
post_install do |installer|
  installer.pods_project.build_configurations.each do |config|
    config.build_settings["BUILD_LIBRARY_FOR_DISTRIBUTION"] = "NO"
  end

  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings["BUILD_LIBRARY_FOR_DISTRIBUTION"] = "NO"
    end
  end
end
```
I guess I'll have to live with this 😅

**Comment by goergisn** [comment]
FYI @norbertlorinczi @paillou 
we just released [5.5.0](https://github.com/Adyen/adyen-ios/releases/tag/5.5.0) that adds support for Library Evolution.
Please let us know if you run into any issues still 👍
